### PR TITLE
Added Order Functionality

### PR DESCRIPTION
**Pull request by DmitryLazyuka**

### CODE REVIEW

File: src/main/java/org/example/springproject/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                                        @PathVariable Long id) {
+        User user = (User) authentication.getPrincipal();
+        return orderService.getSpecificItem(user.getId(), orderId, id);
+    }
+
+    @Operation(summary = "Update Order Status")
+    @PreAuthorize("hasRole('ADMIN')")
+    @PatchMapping("/{orderId}")
+    @ResponseStatus(HttpStatus.OK)
+    OrderDto updateOrderStatus(@RequestBody StatusUpdateRequestDto request,
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/springproject/controller/OrderController.java]:
```suggestion
    public OrderDto updateOrderStatus(@RequestBody StatusUpdateRequestDto request,
```

File: src/main/java/org/example/springproject/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                               @PathVariable Long orderId) {
+        return orderService.updateOrderStatus(orderId, request);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/springproject/controller/OrderController.java:76]: rename request

File: src/main/java/org/example/springproject/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartService shoppingCartService;
+
+    @Override
+    public List<OrderDto> getOrders(Long userId) {
+        List<Order> orders = orderRepository.findByUserId(userId);
+        return orders.stream().map(orderMapper::toDto).toList();
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/springproject/service/impl/OrderServiceImpl.java]:
```suggestion
        return orders.stream()
        .map(orderMapper::toDto)
        .toList();
```

File: src/main/java/org/example/springproject/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Override
+    @Transactional
+    public OrderDto placeOrder(Long userId, PlaceOrderRequestDto requestDto) {
+        ShoppingCart cart = shoppingCartRepository.findByUserId(userId);
+        if (cart == null || cart.getCartItems().isEmpty()) {
+            throw new IllegalStateException("Cart is empty or not available");
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/springproject/service/impl/OrderServiceImpl.java]:
```suggestion
            throw new OrderProcessingException("Cart is empty or not available");
```

File: src/main/java/org/example/springproject/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        order.setShippingAddress(requestDto.getShippingAddress());
+        orderRepository.save(order);
+        shoppingCartService.clearCart(userId);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderItemDto> getOrderItems(Long userId, Long orderId) {
+        Order order = orderRepository.findByIdAndUserId(orderId, userId)
+                .orElseThrow(() -> new NoSuchElementException("Order with " + orderId
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/springproject/service/impl/OrderServiceImpl.java]:
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Order with " + orderId
```

check other places and fix

File: src/main/java/org/example/springproject/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                        + " orderId is not found for current user"));
+        return order.getOrderItems().stream().map(orderMapper::toOrderItemDto).toList();
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/springproject/service/impl/OrderServiceImpl.java]:
```suggestion
        return order.getOrderItems()
        .stream()
        .map(orderMapper::toOrderItemDto)
        .toList();
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/org/example/springproject/exception/CustomGlobalExceptionHandler.java [hunk 7]
```diff
@@ -64,4 +64,13 @@ public ResponseEntity<Object> handleRegistrationException(RegistrationException
...
         return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
     }
+
+    @ExceptionHandler(OrderProcessingException.class)
+    public ResponseEntity<Object> handleROrderProcessingException(OrderProcessingException ex) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp",
+                LocalDateTime.now().format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT)));
+        body.put("error", ex.getMessage());
+        return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/springproject/exception/CustomGlobalExceptionHandler.java]: wrong status

File: src/main/java/org/example/springproject/exception/CustomGlobalExceptionHandler.java [hunk 8]
```diff
@@ -46,6 +47,15 @@ public ResponseEntity<Object> handleEntityNotFoundExceptions(EntityNotFoundExcep
         return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
     }
 
+    @ExceptionHandler(NoSuchElementException.class)
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/springproject/exception/CustomGlobalExceptionHandler.java]: Do you need this exception? Of it's the same as EntityNotFoundException, leave only one

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**